### ISSUE TITLE
Top left most monitor gets focus when spawning a new window

### ISSUE THREAD

**Issue by jklll** [awesomeWM/awesome#2052]
Output of `awesome --version`:
```
awesome v4.2 (Human after all)
 â¢ Compiled against Lua 5.3.4 (running with Lua 5.3)
 â¢ D-Bus support: â
 â¢ execinfo support: â
 â¢ xcb-randr version: 1.5
 â¢ LGI version: 0.9.1
```

**How to reproduce the issue:**

Spawn a new window with a 3 monitor setup.
My setup is arranged like so:
output of xrandr -q:
```
Screen 0: minimum 8 x 8, current 3840 x 2160, maximum 32767 x 32767
DVI-D-0 connected 1920x1080+0+1080 (normal left inverted right x axis y axis) 521mm x 293mm
   1920x1080     60.00*+
   1680x1050     59.95  
   1440x900      59.89  
   1280x1024     75.02    60.02  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   640x480       75.00    72.81    59.94  
HDMI-0 connected primary 1920x1080+1920+80 (normal left inverted right x axis y axis) 0mm x 0mm
   1920x1080     60.00*+  60.00    59.94    50.00    50.00    29.97    25.00    23.97    60.00    50.04  
   1600x1200     60.00  
   1440x900      59.89  
   1366x768      59.79  
   1280x1024     75.02  
   1280x800      59.81  
   1280x720     120.00    60.00    60.00    59.94    50.00  
   1024x768     119.99    75.03    70.07    60.00  
   800x600      119.97    75.00    72.19    60.32    56.25  
   720x576       50.00  
   720x480       59.94  
   640x480      120.01    75.00    72.81    59.94  
DP-0 disconnected (normal left inverted right x axis y axis)
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 disconnected (normal left inverted right x axis y axis)
DP-3 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 521mm x 293mm
   1920x1080     60.00*+
   1680x1050     59.95  
   1440x900      59.89  
   1280x1024     75.02    60.02  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   640x480       75.00    72.81    59.94  
DP-4 disconnected (normal left inverted right x axis y axis)
DP-5 disconnected (normal left inverted right x axis y axis)
```

**Actual result:**
When spawning a new window, if there is a window on the screen closest to position (0,0), then that window gets focus instead of the newly created one.  I've tried reorienting the monitors and no matter what, the monitor closest to (0,0) always gets focus if it has a window.

**Expected result:**
The newly spawned window should have focus.

**Comment by psychon** [comment]
I'm baffled...

Does this happen with the default config? Any other modifications? Are you sure that it is "closest to (0,0)" and not, for example, "window which had the focus last" or "window on some specific screen"?

**Comment by jklll** [comment]
I messed around a bit, and it's definitely the screen closest to (0,0).
With this setup:
DVI-D-0 connected 1920x1080+1920+1080 (normal left inverted right x axis y axis) 521mm x 293mm
HDMI-0 connected primary 1920x1080+0+1919 (normal left inverted right x axis y axis) 0mm x 0mm
DP-3 connected 1920x1080+1920+0 (normal left inverted right x axis y axis) 521mm x 293mm

HDMI-0 is the closest to (0,0) so it always gets focus when a new window is created regardless of which window had focus last.
Running:
xrandr --output HDMI-0 --pos 0x1921
Makes DP-3 the closest to (0,0) so it always gets focus.

Config is default.

**Comment by psychon** [comment]
Well, okay, let's try to reproduce this (awesome `v4.2-116-g790a669-dirty`, Lua 5.3.4).
I added the following to the default config and ran awesome via `make && TESTS_SCREEN_SIZE=2000x1100 tests/run.sh /dev/null`. (The factor of `0.5` is there since I do not have enough screen space, but even without it, things seem to work fine).
```lua
gears.timer.start_new(1, function()
	while screen.count() > 0 do
		screen[1]:fake_remove()
	end
	local function fake_add(x, y, width, height)
		local f = 0.5
		screen.fake_add(f*x, f*y, f*width, f*height)
	end
	fake_add(0, 1080, 1920, 1080)
	fake_add(1920, 80, 1920, 1080)
	fake_add(0, 0, 1920, 1080)
end)
```
With this setup, moving the mouse cursor to one of the three screens and pressing mod4+enter opens a new xterm on the top left part of that screen.

In other words: Apparently I cannot reproduce this.
Can you reproduce the problem when you add the above code to `rc.lua` (I would suggest setting `f=1` for you since you seem to actually have that much screen space)? Any other ideas on what I might be missing?

**Comment by psychon** [comment]
Another idea (in addition to the one above): Let's check out of the windows are being focused. Could you add the following to your `rc.lua` and collect its output? For me, this prints things exactly once for each new window and the traceback goes through `awful.rules.apply`, `awful.rules.execute`, `emit_signal`, `awful.ewmh.active` and then `_newindex`.
```Lua
client.connect_signal("focus", function(c)
	print("Focusing client " .. tostring(c) .. " from:")
	print(debug.traceback())
end)
```

**Comment by jklll** [comment]
I got the correct behavior using the fake screens.  After removing the code and restarting awesome, it started working properly again.  I'm not sure why, but new windows are focusing properly now.  Thanks for the help.  I'll leave the issue open if you want to investigate further.

**Comment by psychon** [comment]
So... this only happens right after you start your computer? And (this is a guess) it stops happening when you restart awesome?
How are you configuring the monitors (assign them position and sizes)? Does this happen before awesome starts or only while awesome is already running?

> I'll leave the issue open if you want to investigate further.

Well, a bug is a bug. I just have no idea what could be going on here.
Clients appear one after another and the code tries to focus a new client when it appears. Since nothing in here picks some "random" client based on anything, I cannot see how the behaviour that you are seeing happens. That makes it quite a bit harder to investigate...

**Comment by Elv13** [comment]
Closing due to the `need info` label being set for a year with no activity.
